cpp/src/bitmask/null_mask.cu

```diff
@@ -172,7 +172,7 @@ __global__ void copy_offset_bitmask(bitmask_type *__restrict__ destination,
         destination_word_index + word_index(bit_offset);
     bitmask_type curr_word = source[source_word_index];
     bitmask_type next_word = 0;
-    if (destination_word_index + 1 < number_of_mask_words) {
+    if ((bit_offset % detail::size_in_bits<bitmask_type>()) != 0) {
```

Comment:
This condition might be better

```
    if ((intra_word_index(source_begin_bit) != 0) &&
        (word_index(source_end_bit - source_begin_bit) > 0)) {...
```

Reply:
If we take _source_begin_bit_ & _source_end_bit_, I think the above code is erroneous. Say _source_begin_bit_ = 1 & _source_end_bit_ = 1 + 32 + 1.

To copy the first mask word, we need to read the next word, but to copy the second mask word, we just need to read 1 bit but  the above condition still turns out to be true.

I think the code below will work.
```
if ((intra_word_index(source_begin_bit) != 0) &&
    (word_index(source_end_bit) >
      word_index(source_begin_bit + destination_word_index * detail::size_in_bits<bitmask_type>()))) {
  ...
}
```
Let me know if you have any concerns.